src/System.Collections/tests/Performance/Perf.Dictionary.cs

```diff
@@ -0,0 +1,195 @@
+                dict.Add("key" + i, "value");
+
+            // Actual perf testing
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    retrieved = dict["key1"]; retrieved = dict["key2"]; retrieved = dict["key3"];
+                    retrieved = dict["key4"]; retrieved = dict["key5"]; retrieved = dict["key6"];
+                    retrieved = dict["key7"]; retrieved = dict["key8"]; retrieved = dict["key9"];
+                }
+            }
+
+            // Teardown
+            for (int i = 1; i <= 9; i++)
+                dict.Remove("key" + i);
+        }
+
+        [Benchmark]
+        [MemberData("TestDictionaries")]
+        public void SetItem(Dictionary<string, string> dict)
+        {
+            // Setup
+            for (int i = 1; i <= 9; i++)
+                dict.Add("key" + i, "value");
+
+            // Actual perf testing
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    dict["key1"] = "string"; dict["key2"] = "string"; dict["key3"] = "string";
+                    dict["key4"] = "string"; dict["key5"] = "string"; dict["key6"] = "string";
+                    dict["key7"] = "string"; dict["key8"] = "string"; dict["key9"] = "string";
+                }
+            }
+
+            // Teardown
+            for (int i = 1; i <= 9; i++)
+                dict.Remove("key" + i);
+        }
+
+        [Benchmark]
+        [MemberData("TestDictionaries")]
+        public void GetKeys(Dictionary<string, string> dict)
+        {
+            IEnumerable<string> result;
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                {
+                    result = dict.Keys; result = dict.Keys; result = dict.Keys;
+                    result = dict.Keys; result = dict.Keys; result = dict.Keys;
+                    result = dict.Keys; result = dict.Keys; result = dict.Keys;
+                }
+        }
+
+        [Benchmark]
+        [MemberData("TestDictionaries")]
+        public void TryGetValue(Dictionary<string, string> dict)
+        {
+            // Setup
+            string retrieved;
+            dict.Add("key", "value");
+
+            // Actual perf testing
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    dict.TryGetValue("key", out retrieved); dict.TryGetValue("key", out retrieved);
+                    dict.TryGetValue("key", out retrieved); dict.TryGetValue("key", out retrieved);
+                    dict.TryGetValue("key", out retrieved); dict.TryGetValue("key", out retrieved);
+                    dict.TryGetValue("key", out retrieved); dict.TryGetValue("key", out retrieved);
```

Comment:
Something to consider with these dictionary tests is how GetHashCode/Equals impacts the perf of the type.  Using a short constant string like this is likely to minimize the time involved in executing those operations, which Dictionary does a lot.  It could be useful to have a test that measure dictionary perf on one or more common operations using some type for which computing these is more expensive, so as to help catch any regressions where a change in dictionary's implementation causes them to be evaluated more often than necessary.